library/kani_macros/src/sysroot/contracts/replace.rs

```diff
@@ -70,6 +70,7 @@ impl<'a> ContractConditionsHandler<'a> {
     /// `use_nondet_result` will only be true if this is the first time we are
     /// generating a replace function.
     fn expand_replace_body(&self, before: &[Stmt], after: &[Stmt]) -> TokenStream {
+        let redefs = self.arg_redefinitions(false);
```

Comment:
Can we do this redefinition just once when we first create the replace body instead of doing it for every contract clause?